python_modules/dagster/dagster/core/system_config/objects.py

```diff
@@ -152,18 +167,28 @@ def build(
         run_config_schema = pipeline_def.get_run_config_schema(mode)
 
         if run_config_schema.config_mapping:
-            outer_evr = process_config(
-                run_config_schema.config_mapping.config_schema.config_type,
-                run_config,
-            )
+            if run_config_schema.config_mapping.enable_pre_processing:
```

Comment:
if we are putting this on `ConfigMapping` I am guessing we are not respecting this in other places that config mappings are evaluated.

likely makes sense to add a method on to `ConfigMapping` that handles this pattern

Reply:
Do you mean a method on ConfigMapping that resolves its own config? Like `ConfigMapping.resolve_from_outer` or something?

Reply:
yeah to perform the operation `ConfigMapping` exists to do, `ConfigMapping.map_config` though that name feels a little rough `ConfigMapping.map`, `ConfigMapping.process`, `ConfigMapping.resolve`, `ConfigMapping.execute`

Reply:
Ended up going with `ConfigMapping.resolve`